include/pinocchio/algorithm/model.hxx

```diff
@@ -256,9 +274,28 @@ namespace pinocchio
 
     geomModel.geometryObjects.reserve (geomModelA.ngeoms + geomModelB.ngeoms);
 
-    // Copy modelA joints until frame.parentJoint
     details::appendUniverseToModel (modelA, geomModelA, 0, id, model, geomModel);
+    // Compute joints of A that should be added before and after joints of B
+    std::vector<JointIndex> AJointsBeforeB;
+    std::vector<JointIndex> AJointsAfterB;
+    // All joints until the parent of frameInModelA come first
     for (JointIndex jid = 1; jid <= frame.parent; ++jid)
+    {
+      AJointsBeforeB.push_back(jid);
+    }
+    // descendants of the parent of frameInModelA come also before model B
+    for (JointIndex jid = frame.parent+1; jid < modelA.joints.size(); ++jid)
+    {
+      if (hasAncestor(modelA, jid, frame.parent))
+      {
+        AJointsBeforeB.push_back(jid);
+      } else
+      {
+        AJointsAfterB.push_back(jid);
+      }
+    }
+    // Copy modelA joints that should come before model B
+    for (auto jid : AJointsBeforeB)
```

Comment:
This is a C++11 feature, targeted for Pinocchio 3.

Reply:
done